x-pack/plugins/dataset_quality/public/hooks/formatted_time.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { useMemo } from 'react';
+import { useKibanaUiSetting } from './use_kibana_ui_setting';
+
+const getFormattedTime = (
```

Comment:
Do we have common time formatters in kibana? is it every screen is formatting the date as they want or in a custom way? Could we at least have a common formatter for observability?

Reply:
yes, I saw we are getting uiSettings my question is more aligned to whether repeating this logic everywhere in the product after getting the uiSetting. Are we sure there is nothing out there already doing this?

Reply:
we could use `fieldFormats` plugin as referenced in this [internal conversation](https://elastic.slack.com/archives/C0D8P2XK5/p1700740158418339?thread_ts=1700738930.498019&cid=C0D8P2XK5).